plugins/chanbackup.c

```diff
@@ -362,6 +495,237 @@ static struct command_result *json_state_changed(struct command *cmd,
+
+                req = jsonrpc_request_start(cmd->plugin,
+                                    cmd,
+                                    "datastore",
+                                    after_datastore,
+                                    &forward_error,
+                                    NULL);
+
+                json_array_start(req->js, "key");
+                json_add_string(req->js, NULL, "latestbkp");
+                json_array_end(req->js);
+                json_add_hex(req->js, "hex", decoded_bkp,
+                             tal_bytelen(decoded_bkp));
+                json_add_string(req->js, "mode", "create-or-replace");
+
+                return send_outreq(cmd->plugin, req);
+        }
+
+        req = jsonrpc_request_start(cmd->plugin,
+                                    cmd,
+                                    "datastore",
+                                    after_datastore,
+                                    &forward_error,
+                                    NULL);
+
+        json_array_start(req->js, "key");
+        json_add_node_id(req->js, NULL, node_id);
+        json_add_string(req->js, NULL, "chanbackup");
+	json_array_end(req->js);
+        json_add_hex(req->js, "hex", payload_deserialise,
+                     tal_bytelen(payload_deserialise));
+        json_add_string(req->js, "mode", "create-or-replace");
+
+        return send_outreq(cmd->plugin, req);
+}
+
+static struct command_result *after_latestbkp(struct command *cmd,
+					        const char *buf,
+					        const jsmntok_t *params,
+					        void *cb_arg UNUSED)
+{
+        u64 version;
+	u32 timestamp;
+	struct scb_chan **scb;
+        struct json_stream *response;
+        const jsmntok_t *datastoretok = json_get_member(buf,
+                                                        params,
+                                                        "datastore");
+        struct out_req *req;
+
+        if (datastoretok->size == 0) {
+        	response = jsonrpc_stream_success(cmd);
+		json_add_string(response, NULL,
+                                "No backup received from peers");
+		return command_finished(cmd, response);
+        }
+
+        const jsmntok_t *bkp = json_get_member(buf,
+                                               json_get_arr(datastoretok, 0),
+                                               "hex");
+
+        u8 *res = tal_hexdata(cmd,
+                              json_strdup(cmd, buf, bkp),
+                              bkp->end - bkp->start);
+	if (!fromwire_static_chan_backup(cmd,
+                                         res,
+                                         &version,
+                                         &timestamp,
+                                         &scb)) {
+		plugin_err(cmd->plugin, "Corrupted SCB!");
+	}
+
+	if (version != VERSION) {
+		plugin_err(cmd->plugin,
+                           "Incompatible version, Contact the admin!");
```

Comment:
who's the `admin`? 

`incompatible scb file version on disk`.

Reply:
> who's the `admin`?
> you

Reply:
If they have an older version, they can contact the team to see if it could be parsed and used for fund recovery.